### ISSUE TITLE
ToMatchSnapshot broken with react components

### ISSUE THREAD

**Issue by The-Code-Monkey** [oven-sh/bun#5540]
### What version of Bun is running?

latest

### What platform is your computer?

Mac osx latest M2 chip

### What steps can reproduce the bug?

Create a simple react component and render it in a test using @testing-library/react 

```
const Text = ({ text }) => <>{text}</>;
const LazyText = await lazy(() => fakeImport(Text), 'test');

    const { asFragment } = render(
      <Suspense fallback={<Text text='Loading...' />}>
        <LazyText text='hi' />
      </Suspense>
    );

    expect(asFragment()).toMatchSnapshot();
```

### What is the expected behavior?

It should print out the html into a snapshot file

### What do you see instead?

Instead it prints out a massive JSON object which in one case was over 20mb this cannot be correct

### Additional information

Sorry did this on my mobile was unable to get specific values for the versions.

One thing i forgot to add is that i am using happy dom

**Comment by Electroid** [comment]
May be related to:
* #5272 
* #3769

**Comment by The-Code-Monkey** [comment]
Do we know if this is being looked at? And is there a time frame for a fix?

Or is this something that supporting jsdom would fix as this is what I used
previously with jest

**Comment by The-Code-Monkey** [comment]
@Electroid I have found the fix by trawling the Jest code,

at some point jest calls a serialize function which runs this bit of code.

```
prettyFormat(val, {
      escapeRegex: true,
      indent: 2,
      plugins: PLUGINS,
      printFunctionName: false
    })
```

This code along with the attached plugins makes the horrible json object become 

```
<DocumentFragment>
  Loading...
</DocumentFragment>
```

I'm not sure how we would bake this in unless there was a way to add functionality to the expect function via the setup file

**Comment by colinhacks** [comment]
Here's a way to switch from Jest to Bun when relying on React Testing Library snapshots. It manually calls the `prettyFormat` function from Jest's `pretty-format` module, which is used by Jest internally.

```diff
  import { test, describe, expect } from "bun:test";
  import {
    format as prettyFormat,
    plugins as prettyFormatPlugins,
  } from "pretty-format";

  function format(value: any) {
    return prettyFormat(value, {
      escapeRegex: true,
      indent: 2,
      plugins: [
        prettyFormatPlugins.ReactTestComponent,
        prettyFormatPlugins.ReactElement,
        prettyFormatPlugins.DOMElement,
        prettyFormatPlugins.DOMCollection,
        prettyFormatPlugins.Immutable,
        prettyFormatPlugins.AsymmetricMatcher,
      ],
      printFunctionName: false,
    });
  }

  test("test", () => {
    const frag = render(<Message message="hello world" />);
-   expect(frag.asFragment()).toMatchSnapshot();
+   expect(format(frag.asFragment())).toMatchSnapshot();
  })
```

**Comment by The-Code-Monkey** [comment]
Yeah i found this solution but it's a bit annoying having to do this for
every project it would be nice if this was baked in like it is for jest

**Comment by LukaszKosobucki** [comment]
Could this be included in a configuration file, such as the RTL config, to make it applicable project-wide? This way, we wouldn’t need to create utility functions and import them everywhere.

I also agree that it should be natively integrated into Bun. Without this, Bun tests freeze on .toMatchSnapshot() when any component includes curly braces expressions. (at least for me)

also thanks for working workaround.
